lib/rules/autocomplete-matches.js

```diff
@@ -1,31 +1,34 @@
 const { text, aria, dom } = axe.commons;
 
-const autocomplete = node.getAttribute('autocomplete');
+const autocomplete = virtualNode.attr('autocomplete');
 if (!autocomplete || text.sanitize(autocomplete) === '') {
 	return false;
 }
 
-const nodeName = node.nodeName.toUpperCase();
-if (['TEXTAREA', 'INPUT', 'SELECT'].includes(nodeName) === false) {
+const nodeName = virtualNode.elementNodeName;
+if (['textarea', 'input', 'select'].includes(nodeName) === false) {
 	return false;
 }
 
 // The element is an `input` element a `type` of `hidden`, `button`, `submit` or `reset`
 const excludedInputTypes = ['submit', 'reset', 'button', 'hidden'];
-if (nodeName === 'INPUT' && excludedInputTypes.includes(node.type)) {
+if (
```

Comment:
`hidden` inputs can still have `autocomplete`, is this exclusion based on some other factor?